### PR TITLE
Grouper support

### PR DESCRIPTION
**Pull request by ajragusa**
This should complete implementation of #743 Grouper support in OESS

### CODE REVIEW

File: app/grouper_syncer.pl [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+#      "employeenumber",
+#      "cn",
+#      "mail"
+#    ]
+#  }
+#}';
+	
+	my $h = HTTP::Headers->new(
+	    Content_Length => length($str),
+	    Content_Type => 'text/x-json' );
```
Comment (jonstout) [inline app/grouper_syncer.pl]: `text/x-json`, Is that right?

File: app/grouper_syncer.pl [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+#      "givenname",
+#      "sn",
+#      "edupersonprincipalname",
+#      "uid",
+#      "employeenumber",
+#      "cn",
+#      "mail"
+#    ]
+#  }
+#}';
```
Comment (jonstout) [inline app/grouper_syncer.pl]: Would probably make more sense to format the actual request and remove this comment?

File: app/grouper_syncer.pl [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+#  "WsRestFindGroupsRequest":{
+#    "actAsSubjectLookup":{
+#      "subjectId":"oess_provision"
+#    },
+#    "wsQueryFilter":{
+#      "queryFilterType":"FIND_BY_STEM_NAME",
+#      "stemName":"app:oess:service:workgroups"
+#    }
+#  }
+#}';
```
Comment (jonstout) [inline app/grouper_syncer.pl]: Remove comment. Could be nice to have the stemName included in an inline comment as an example.

File: app/grouper_syncer.pl [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+    my $h = HTTP::Headers->new(
+	Content_Length => length($str),
+	Content_Type => 'text/x-json' );
```
Comment (jonstout) [inline app/grouper_syncer.pl]: Correct content type?

File: app/grouper_syncer.pl (hunk 1 above)
Comment (ajragusa) [reply app/grouper_syncer.pl]: yes actually this is required by the Grouper API

File: app/grouper_syncer.pl (hunk 2 above)
Comment (ajragusa) [reply app/grouper_syncer.pl]: ya removed

File: app/grouper_syncer.pl (hunk 4 above)
Comment (ajragusa) [reply app/grouper_syncer.pl]: Yes actually this is!  Its required by the Grouper API

File: app/grouper_syncer.pl (hunk 3 above)
Comment (ajragusa) [reply app/grouper_syncer.pl]: Should be fixed now!

File: app/grouper_syncer.pl [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+$ENV{HTTPS_DEBUG} = 0;
+
+my %all_oess_users;
+
+sub main{
+    my ($config_file, $verbose) = @_;
+
+    #init_log4perl();
+
+    $log = GRNOC::Log->new( level => 'INFO' );
```
[the remark this answers is not in the document]
Comment (ajragusa) [reply app/grouper_syncer.pl]: Ya I don't really need $log as I am treating it like a singleton... doesn't hurt though incase I wanted to not do that :)

File: app/grouper_syncer.pl [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+    
+    return \@grouper_users;
+}
+
+
+
+sub fetch_config{
+    my $config_file = shift;
+    log_debug("Config File: " . $config_file); 
+    my $config = GRNOC::Config->new( config_file => $config_file);
```
[the remark this answers is not in the document]
Comment (ajragusa) [reply app/grouper_syncer.pl]: should be fixed now

**Comment by jonstout** [comment]
rebuild

**Pull request merged**